### PR TITLE
Added ctypes wrapper for krb5_host_realm.

### PR DESCRIPTION
**Pull request by andreikeis**

### CODE REVIEW

File: lib/python/treadmill/syscall/krb5.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+        return None
+    if _KRB5_FREE_HOST_REALM is None:
+        return None
+
+    # krb5_error_code retval = 0;
+    # krb5_context kcontext;
+    #
+    # retval = krb5_init_context(&kcontext);
+    kcontext = c_void_p()
+    krb5_error_code = _KRB5_INIT_CONTEXT(byref(kcontext))
```
Comment (ceache) [inline lib/python/treadmill/syscall/krb5.py]: "The context must be released by calling krb5_free_context() when it is no longer needed." from the doc

File: lib/python/treadmill/syscall/krb5.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    while True:
+        realm = realm_list[idx]
+        if not realm:
+            break
+
+        realms.append(realm.decode())
+        idx = idx + 1
+
+    # krb5_free_host_realm(kcontext, realm_list);
+    krb5_error_code = _KRB5_FREE_HOST_REALM(kcontext, realm_list)
```
Comment (ceache) [inline lib/python/treadmill/syscall/krb5.py]: I would wrap all the frees in a try...finally statement, to be safe.

File: lib/python/treadmill/syscall/krb5.py (hunk 1 above)
Comment (andreikeis) [reply lib/python/treadmill/syscall/krb5.py]: Fixed.

File: lib/python/treadmill/syscall/krb5.py (hunk 2 above)
Comment (andreikeis) [reply lib/python/treadmill/syscall/krb5.py]: fixed.

**Pull request merged**